### PR TITLE
Create chat controller and hub

### PR DESCRIPTION
**Pull request by Ovsyannikov345**

### CODE REVIEW

File: ChatService/ChatService.BLL/Services/ChatService.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+            chatModel.User = new UserProfileModel
+            {
+                Id = Guid.Parse(profileResponse.Profile.Id),
+                Auth0Id = profileResponse.Profile.Auth0Id,
+                FirstName = profileResponse.Profile.FirstName,
+                LastName = profileResponse.Profile.LastName,
+                PhoneNumber = profileResponse.Profile.PhoneNumber,
+                BirthDate = profileResponse.Profile.BirthDate.ToDateTime()
+            };
```
Comment (DmitriyG0r4anin) [inline ChatService/ChatService.BLL/Services/ChatService.cs]: Why not to use AutoMapper instead?

**Review by DmitriyG0r4anin — APPROVED** [review]

File: ChatService/ChatService.BLL/Models/Enums/EstateType.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿namespace ChatService.BLL.Models.Enums
+{
+    public enum EstateType
```
Comment (nantonov) [inline ChatService/ChatService.BLL/Models/Enums/EstateType.cs:3]: add default value?

File: ChatService/ChatService.BLL/Models/MessageModel.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    {
+        public Guid Id { get; set; }
+
+        public string Text { get; set; } = string.Empty;
+
+        public DateTime CreatedAt { get; set; } = DateTime.Now;
+
+        public Guid ChatId { get; set; }
+
+        public ChatModel Chat { get; set; } = null!;
```
Comment (nantonov) [inline ChatService/ChatService.BLL/Models/MessageModel.cs]: mark as required?

File: ChatService/ChatService.BLL/Models/UserProfileModel.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿namespace ChatService.BLL.Models
+{
+    public class UserProfileModel
+    {
+        public Guid Id { get; set; }
+
+        public string Auth0Id { get; set; } = string.Empty;
```
Comment (nantonov) [inline ChatService/ChatService.BLL/Models/UserProfileModel.cs]: reuired? and following

File: ChatService/ChatService.BLL/Services/ChatService.cs (hunk 1 above)
Comment (nantonov) [reply ChatService/ChatService.BLL/Services/ChatService.cs]: +

File: ChatService/ChatService.BLL/Services/IServices/IChatService.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+﻿using ChatService.BLL.Models;
+
+namespace ChatService.BLL.Services.IServices
+{
+    public interface IChatService
+    {
+        public Task<ChatModel> GetChatAsync(Guid id, CancellationToken cancellationToken = default);
```
Comment (nantonov) [inline ChatService/ChatService.BLL/Services/IServices/IChatService.cs]: del public

File: ChatService/ChatService.BLL/Services/IServices/IMessageService.cs [hunk 6]
```diff
@@ -0,0 +1,9 @@
+﻿using ChatService.BLL.Models;
+
+namespace ChatService.BLL.Services.IServices
+{
+    public interface IMessageService
+    {
+        public Task<MessageModel> CreateAsync(MessageModel messageData, CancellationToken cancellationToken = default);
```
Comment (nantonov) [inline ChatService/ChatService.BLL/Services/IServices/IMessageService.cs]: del public

File: ChatService/ChatService.DAL/Models/Entities/Message.cs [hunk 7]
```diff
@@ -0,0 +1,18 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace ChatService.DAL.Models.Entities
+{
+    public class Message
+    {
+        public Guid Id { get; set; }
+
+        [MaxLength(300)]
+        public string Text { get; set; } = string.Empty;
```
Comment (nantonov) [inline ChatService/ChatService.DAL/Models/Entities/Message.cs]: required again?

File: ChatService/ChatService.DAL/Models/Entities/Message.cs [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+        public DateTime CreatedAt { get; set; } = DateTime.Now;
```
Comment (nantonov) [inline ChatService/ChatService.DAL/Models/Entities/Message.cs]: why not UTC in this case?

**Review by nantonov — CHANGES_REQUESTED** [review]

File: ChatService/ChatService.BLL/Services/ChatService.cs (hunk 1 above)
Comment (VadimZharikov) [reply ChatService/ChatService.BLL/Services/ChatService.cs]: +1

**Review by VadimZharikov — APPROVED** [review]

File: ChatService/ChatService.BLL/Models/Enums/EstateType.cs (hunk 2 above)
Comment (Ovsyannikov345) [reply ChatService/ChatService.BLL/Models/Enums/EstateType.cs:3]: The default value for EstateType right now is Apartment. There is no need to create some kind of unspecified value in my opinion. I can explicitly assign EstateType Type = EstateType.Apartment in the EstateModel if necessary

File: ChatService/ChatService.BLL/Models/MessageModel.cs (hunk 3 above)
Comment (Ovsyannikov345) [reply ChatService/ChatService.BLL/Models/MessageModel.cs]: This field is reduntant, removed it

File: ChatService/ChatService.BLL/Services/ChatService.cs (hunk 1 above)
Comment (Ovsyannikov345) [reply ChatService/ChatService.BLL/Services/ChatService.cs]: Fixed .proto files namespaces and created second map in order to use mapper here

**Review by nantonov — APPROVED** [review]

**Pull request merged**